<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to rendseq’s documentation! &mdash; rendseq 2022 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="#" class="icon icon-home"> rendseq
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Welcome to rendseq’s documentation!</a></li>
<li><a class="reference internal" href="#module-rendseq.file_funcs">File Functions</a></li>
<li><a class="reference internal" href="#module-rendseq.zscores">Functions for Calculating Z-Scores</a></li>
<li><a class="reference internal" href="#module-rendseq.make_peaks">Functions for calling peaks from Z-Scores</a></li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a></li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">rendseq</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
      <li>Welcome to rendseq’s documentation!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="welcome-to-rendseq-s-documentation">
<h1>Welcome to rendseq’s documentation!<a class="headerlink" href="#welcome-to-rendseq-s-documentation" title="Permalink to this headline"></a></h1>
<p>This is an in-development package for the analysis of end-enriched RNA sequencing data.</p>
<p>For more information, see our GitHub page: <a class="reference external" href="https://github.com/miraep8/rendseq">https://github.com/miraep8/rendseq</a></p>
</section>
<section id="module-rendseq.file_funcs">
<span id="file-functions"></span><h1>File Functions<a class="headerlink" href="#module-rendseq.file_funcs" title="Permalink to this headline"></a></h1>
<p>Functions for fetching, creating, and opening raw and processed data files.</p>
<dl class="py function">
<dt class="sig sig-object py" id="rendseq.file_funcs.make_new_dir">
<span class="sig-prename descclassname"><span class="pre">rendseq.file_funcs.</span></span><span class="sig-name descname"><span class="pre">make_new_dir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dir_parts</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.file_funcs.make_new_dir" title="Permalink to this definition"></a></dt>
<dd><p>Create a new directory and return valid path to it.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>name</strong> (<em>- dir_parts  - a list of strings to be joined to make the directory</em>) – </p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p><ul class="simple">
<li><p>dir_str - the directory name</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.file_funcs.open_wig">
<span class="sig-prename descclassname"><span class="pre">rendseq.file_funcs.</span></span><span class="sig-name descname"><span class="pre">open_wig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.file_funcs.open_wig" title="Permalink to this definition"></a></dt>
<dd><p>Open the provided wig file and return the contents into a 2xn array.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>required</strong> (<em>-filename</em><em> (</em><em>string</em><em>) </em><em>-</em>) – the filename you desire to open!</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><strong>-reads (2xn array)</strong> – and the second column being the count at that position (raw read,
z_score etc)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>a 2xn array with the first column being position</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.file_funcs.validate_reads">
<span class="sig-prename descclassname"><span class="pre">rendseq.file_funcs.</span></span><span class="sig-name descname"><span class="pre">validate_reads</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reads</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.file_funcs.validate_reads" title="Permalink to this definition"></a></dt>
<dd><p>Make sure the given reads meet our format requirements.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>array</strong><strong>)</strong> (<em>-reads</em><em> (</em><em>2xn</em>) – and the second column being the count at that position (raw read,
z_score etc)</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>NoneType</p>
</dd>
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><strong>Throws exceptions if reads are not correctly formatted</strong> – </p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.file_funcs.write_wig">
<span class="sig-prename descclassname"><span class="pre">rendseq.file_funcs.</span></span><span class="sig-name descname"><span class="pre">write_wig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">wig_track</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">wig_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.file_funcs.write_wig" title="Permalink to this definition"></a></dt>
<dd><p>Write provided data to the wig file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>array</strong><strong>)</strong> (<em>- wig_track</em><em> (</em><em>required</em><em>) </em><em>- the wig data you wish to write</em><em> (</em><em>in 2xn</em>) – </p></li>
<li><p><strong>to</strong> (<em>- wig_file_name</em><em> (</em><em>string</em><em>) </em><em>- the new file you will write</em>) – </p></li>
</ul>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-rendseq.zscores">
<span id="functions-for-calculating-z-scores"></span><h1>Functions for Calculating Z-Scores<a class="headerlink" href="#module-rendseq.zscores" title="Permalink to this headline"></a></h1>
<p>Functions needed for z-score transforming raw rendSeq data.</p>
<dl class="py function">
<dt class="sig sig-object py" id="rendseq.zscores.main_zscores">
<span class="sig-prename descclassname"><span class="pre">rendseq.zscores.</span></span><span class="sig-name descname"><span class="pre">main_zscores</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.zscores.main_zscores" title="Permalink to this definition"></a></dt>
<dd><p>Run Z-score calculations.</p>
<p>Effect: Writes messages to standard out. If –save-file flag,
also writes output to disk.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.zscores.parse_args_zscores">
<span class="sig-prename descclassname"><span class="pre">rendseq.zscores.</span></span><span class="sig-name descname"><span class="pre">parse_args_zscores</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">args</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.zscores.parse_args_zscores" title="Permalink to this definition"></a></dt>
<dd><p>Parse command line arguments.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.zscores.score_helper">
<span class="sig-prename descclassname"><span class="pre">rendseq.zscores.</span></span><span class="sig-name descname"><span class="pre">score_helper</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stop</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_r</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reads</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">i</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.zscores.score_helper" title="Permalink to this definition"></a></dt>
<dd><p>Find the z-score of reads[i] relative to the subsection of reads.</p>
<p>Goes from start to stop, with a read cutoff of min_r</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.zscores.validate_gap_window">
<span class="sig-prename descclassname"><span class="pre">rendseq.zscores.</span></span><span class="sig-name descname"><span class="pre">validate_gap_window</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gap</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">w_sz</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.zscores.validate_gap_window" title="Permalink to this definition"></a></dt>
<dd><p>Check that gap and window size are reasonable in r/l_score_helper.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.zscores.z_score">
<span class="sig-prename descclassname"><span class="pre">rendseq.zscores.</span></span><span class="sig-name descname"><span class="pre">z_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">val</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">v_mean</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">v_std</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.zscores.z_score" title="Permalink to this definition"></a></dt>
<dd><p>Calculate a z-score given a value, mean, and standard deviation.</p>
<p>NOTE: The z_score() of a constant vector is 0</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.zscores.z_scores">
<span class="sig-prename descclassname"><span class="pre">rendseq.zscores.</span></span><span class="sig-name descname"><span class="pre">z_scores</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reads</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">w_sz</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_r</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.zscores.z_scores" title="Permalink to this definition"></a></dt>
<dd><p>Perform modified z-score transformation of reads.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>reads</strong> (<em>-reads 2xn array - raw rendseq</em>) – </p></li>
<li><p><strong>(</strong><strong>interger</strong><strong>)</strong> (<em>-gap</em>) – interest that should be excluded in the z_score calculation.</p></li>
<li><p><strong>(</strong><strong>integer</strong><strong>)</strong> (<em>-min_r</em>) – one should include in zscore calulcation.</p></li>
<li><p><strong>(</strong><strong>integer</strong><strong>)</strong> – of reads going into the z_score calculation for a point that point
is excluded.  note this is sum of reads in the window</p></li>
<li><p><strong>(</strong><strong>string</strong><strong>)</strong> (<em>-file_name</em>) – the message printed</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><strong>-z_score (2xn array)</strong> – and the second column being the z_score.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>a 2xn array with the first column being position</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-rendseq.make_peaks">
<span id="functions-for-calling-peaks-from-z-scores"></span><h1>Functions for calling peaks from Z-Scores<a class="headerlink" href="#module-rendseq.make_peaks" title="Permalink to this headline"></a></h1>
<p>Take normalized raw data find the peaks in it.</p>
<dl class="py function">
<dt class="sig sig-object py" id="rendseq.make_peaks.hmm_peaks">
<span class="sig-prename descclassname"><span class="pre">rendseq.make_peaks.</span></span><span class="sig-name descname"><span class="pre">hmm_peaks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">z_scores</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">i_to_p</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_to_p</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.6666666666666666</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">peak_center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">spread</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.make_peaks.hmm_peaks" title="Permalink to this definition"></a></dt>
<dd><p>Fit peaks to the provided z_scores data set using the vertibi algorithm.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>array</strong><strong>)</strong> (<em>-z_scores</em><em> (</em><em>2xn</em>) – location) second column is a modified z_score for that position.</p></li>
<li><p><strong>(</strong><strong>float</strong><strong>)</strong> (<em>-spread</em>) – probability of transitioning from inernal state to peak state. The
default value is 1/2000, based on asseumption of geometrically
distributed transcript lengths with mean length 2000. Should be a
robust parameter.</p></li>
<li><p><strong>(</strong><strong>float</strong><strong>)</strong> – 1/1.5.</p></li>
<li><p><strong>(</strong><strong>float</strong><strong>)</strong> – for the peak state.</p></li>
<li><p><strong>(</strong><strong>float</strong><strong>)</strong> – distribution.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><strong>-peaks</strong> – column being a peak assignment.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>a 2xn array with the first column being position and the second</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.make_peaks.main_make_peaks">
<span class="sig-prename descclassname"><span class="pre">rendseq.make_peaks.</span></span><span class="sig-name descname"><span class="pre">main_make_peaks</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.make_peaks.main_make_peaks" title="Permalink to this definition"></a></dt>
<dd><p>Run the main peak making from command line.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.make_peaks.parse_args_make_peaks">
<span class="sig-prename descclassname"><span class="pre">rendseq.make_peaks.</span></span><span class="sig-name descname"><span class="pre">parse_args_make_peaks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">args</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.make_peaks.parse_args_make_peaks" title="Permalink to this definition"></a></dt>
<dd><p>Parse command line arguments.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="rendseq.make_peaks.thresh_peaks">
<span class="sig-prename descclassname"><span class="pre">rendseq.make_peaks.</span></span><span class="sig-name descname"><span class="pre">thresh_peaks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">z_scores</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">thresh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'kink'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#rendseq.make_peaks.thresh_peaks" title="Permalink to this definition"></a></dt>
<dd><p>Find peaks by calling z-scores above a threshold as a peak.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pos.</strong> (<em>- z_scores - a 2xn array of nt positions and zscores at that</em>) – </p></li>
<li><p><strong>be</strong> (<em>- thresh - the threshold value to use.  If none is provided it will</em>) – automatically calculated.</p></li>
<li><p><strong>score</strong> (<em>- method - the method to use to automatically calculate the z</em>) – if none is provided.  Default method is “kink”</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<div class="toctree-wrapper compound">
</div>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Mirae Parker.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>